crates/dbsp/src/circuit/runtime.rs

```diff
@@ -16,16 +19,28 @@ use std::{
 };
 use typedmap::{TypedDashMap, TypedMapKey};
 
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize)]
 pub enum Error {
-    WorkerPanic(usize),
+    /// Panic in a worker thread.
+    WorkerPanic {
+        worker: usize,
+    },
     Killed,
 }
 
+impl DetailedError for Error {
+    fn error_code(&self) -> Cow<'static, str> {
+        match self {
+            Self::WorkerPanic { .. } => Cow::from("WorkerPanic"),
+            Self::Killed => Cow::from("Killed"),
```

Comment:
Shutdown/terminated, as that's the wording used elsewhere. Also less violent. :)